string[] action
string[] obj1
string[] obj2
string[] dir
string[] dis
